src/parakeet/Pro/Driver.cpp

```diff
@@ -432,12 +337,12 @@ namespace Pro
                 }
 
                 ScanData* data = new ScanData;
```

Comment:
You currently delete the ScanData* object inside the onScanDataReceived method so it seems like you don't really need a pointer here if you change the onScanDataReceived as I suggest and pass a const ScanData& instead of transferring ownership.